### PR TITLE
rng_host_guest_read:Host and guest read data at the same time

### PR DESCRIPTION
**Pull request by lijinlijin**
Add a new case to support host and guest read random numbers at
the same time.

Signed-off-by: Li Jin <lijin@redhat.com>

ID:1496724

### CODE REVIEW

**Comment by lijinlijin** [comment]
'1-smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u4.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.rng_host_guest_read' Finshed 2017-10-12 01:26:07 2017-10-12 01:30:42 PASS 274.730956'
1-smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u4.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.rng_host_guest_read' Finshed 2017-10-12 01:33:15 2017-10-12 01:37:49 PASS 273.324101

**Comment by lijinlijin** [comment]
@suqinhuang @hereischen 
I open this new PR,please have a check,thanks

File: qemu/tests/rng_host_guest_read.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                 else return False
+        """
+        error_context.context("Check whether host is reading is alive",
+                              logging.info)
+        host_read_check_cmd = params.get("host_read_check_cmd")
+        output = process.system_output(host_read_check_cmd, shell=True)
+        result = re.findall(target_process, output, re.M | re.I)
+        return bool(result)
+
+    def host_read_stop(host_read_process):
```
Comment (hereischen) [inline qemu/tests/rng_host_guest_read.py]: I think this function is a little bit overkill.
I know this probably will not happen, but what if the process were stopped before executing  `host_read_process.stop()`

Comment (lijinlijin) [reply qemu/tests/rng_host_guest_read.py]: Modified.
Add a judgement condition to check if the host_read_process is still alive.
If yes,call stop() function
Thanks

File: qemu/tests/rng_host_guest_read.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    vm.wait_for_login()
+
+    error_context.context("Host read random numbers in the background",
+                          logging.info)
+    host_read_process = host_read_start(host_read_cmd)
+
+    if is_host_read_running(host_read_cmd):
+        error_context.context("Guest begin to read random numbers",
+                              logging.info)
+        utils_test.run_virt_sub_test(test, params, env, rng_test)
```
Comment (xiagao) [inline qemu/tests/rng_host_guest_read.py]: Is it better to change parameter rng_test's name to guest related for this case?
such as guest_read_test or guest_rng_test

Comment (lijinlijin) [reply qemu/tests/rng_host_guest_read.py]: Modified.
Thanks

File: qemu/tests/cfg/rng_host_guest_read.cfg [hunk 3]
```diff
@@ -0,0 +1,24 @@
+- rng_host_guest_read:
+    type = rng_host_guest_read
+    kill_vm = yes
+    virt_test_type = qemu
+    no no_virtio_rng
+    guest_rng_test = rng_bat
+    host_read_cmd = "dd if=/dev/urandom of=/dev/null bs=100000 count=10000000"
+    host_read_check_cmd = "ps aux | grep 'dd if=/dev/urandom' | grep R+"
```
Comment (luckyh) [inline qemu/tests/cfg/rng_host_guest_read.cfg]: This can be implemented by `process.poll() is None`.

File: qemu/tests/rng_host_guest_read.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    if is_host_read_running(host_read_cmd):
+        error_context.context("Guest begin to read random numbers",
+                              logging.info)
+        utils_test.run_virt_sub_test(test, params, env, guest_rng_test)
+    else:
+        test.fail("Host reading data is not alive!")
+
+    error_context.context("Clean host read process", logging.info)
+    host_read_stop(host_read_process)
```
Comment (luckyh) [inline qemu/tests/rng_host_guest_read.py]: We should terminate or kill the process rather than stop it.

File: qemu/tests/rng_host_guest_read.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    error_context.context("Host read random numbers in the background",
+                          logging.info)
+    host_read_process = host_read_start(host_read_cmd)
+
+    if is_host_read_running(host_read_cmd):
+        error_context.context("Guest begin to read random numbers",
+                              logging.info)
+        utils_test.run_virt_sub_test(test, params, env, guest_rng_test)
+    else:
+        test.fail("Host reading data is not alive!")
```
Comment (luckyh) [inline qemu/tests/rng_host_guest_read.py]: Should be `test.error`.

File: qemu/tests/rng_host_guest_read.py (hunk 4 above)
Comment (luckyh) [reply qemu/tests/rng_host_guest_read.py]: And we should do this action even if the test failed.

**Comment by lijinlijin** [comment]
@luckyh 
All suggestions are updated.
Thanks

PASS 1-smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u4.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.rng_host_guest_read

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**